edward/data.py

```diff
@@ -30,57 +38,71 @@ def __init__(self, data=None, shuffled=True):
             # shuffle self.data
             raise NotImplementedError()
 
-        self.counter = 0
         if self.data is None:
-            self.N = None
+            pass
         elif isinstance(self.data, tf.Tensor):
             self.N = self.data.get_shape()[0].value
+            self.counter = 0
```

Comment:
you have a big data set of size `N`. `counter` tells us where in this data set we currently are to get the next mini-batch of size `n_data`, i.e., data indices `self.counter:(self.counter + n_data)` with the nuance of looping back when `self.counter + n_data > N`.

i will document this